src/webapp/reports/glass-admin/glass-admin-list/atc-classification/useATCUpload.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from "react";
+import { useBooleanState } from "../../../../utils/use-boolean";
+import { useAppContext } from "../../../../contexts/app-context";
+import { AMCRecalculation } from "../../../../../domain/reports/glass-admin/entities/GLASSDataMaintenanceItem";
+import { Namespaces } from "../../../../../data/common/clients/storage/Namespaces";
+
+export function useATCUpload() {
+    const { compositionRoot } = useAppContext();
+
+    const [isPatchModalOpen, { enable: openPatchModal, disable: closePatchModal }] = useBooleanState(false);
+    const [isUploadATCModalOpen, { enable: openUploadATCModal, disable: closeUploadATCModal }] = useBooleanState(false);
+    const [isRecalculateLogicModalOpen, { enable: openRecalculateLogicModal, disable: closeRecalculateLogicModal }] =
+        useBooleanState(false);
+
+    const [recalculationLogic, setRecalculationLogic] = useState<AMCRecalculation>();
+    const [loggerProgram, setLoggerProgram] = useState<string>("");
+
+    useEffect(() => {
```

Comment:
Here instead of first getting data in `amc-recalculation` to just use the `loggerProgram` value to then take the name of the program (which is what is shown), it's better to get in the use case `GetATCLoggerProgramUseCase` the `loggerProgram` from data in `amc-recalculation` and with that get the name of the program which is returned